pkg/kafkautil/healthcheck.go

```diff
@@ -38,3 +40,32 @@ func (k *kafkaClient) OfflineReplicaCount() (int, error) {
 	log.Info(fmt.Sprintf("offline Replica Count is %d", offlineReplicaCount))
 	return offlineReplicaCount, nil
 }
+
+func (k *kafkaClient) AllReplicaInSync() (bool, error) {
+	availableTopics, err := k.client.Topics()
+	if err != nil {
+		return false, err
+	}
+	for _, topic := range availableTopics {
+		partitions, err := k.client.Partitions(topic)
+		if err != nil {
+			return false, err
+		}
+		for _, partition := range partitions {
+			replicas, err := k.client.Replicas(topic, partition)
+			if err != nil {
+				return false, err
+			}
+			isrreplicas, err := k.client.InSyncReplicas(topic, partition)
+			if err != nil {
+				return false, err
+			}
+			if len(replicas) != len(isrreplicas) {
+				log.Info("not all replicas are in sync")
```

Comment:
possibly worth to gather all the out of sync replicas and log which are out of sync? for debugging/troubleshooting...

Reply:
This is actually getting done by the kafkatopic CR itself too for whatever thats worth. At least if the topic was made through that. The status will be kept up to date and detail any out of sync replicas along with partition leaders and such.